grpc/middleware/xray/middleware_test.go

```diff
@@ -52,3 +98,472 @@ func TestNewStreamServer(t *testing.T) {
 		}
 	}
 }
+
+func TestUnaryServerMiddleware(t *testing.T) {
+	var (
+		traceID    = "traceID"
+		spanID     = "spanID"
+		parentID   = "parentID"
+		clientIP   = "104.18.43.42"
+		remoteAddr = "104.18.43.42:443"
+		agent      = "user agent"
+		unary      = &grpc.UnaryServerInfo{FullMethod: "Test.Test"}
+	)
+	cases := map[string]*testCase{
+		"no-trace": {
+			Trace:    Tra{"", "", ""},
+			Request:  Req{"", "", ""},
+			Response: Res{codes.OK},
+			Segment:  Seg{nil, false},
+		},
+		"basic": {
+			Trace:    Tra{traceID, spanID, ""},
+			Request:  Req{remoteAddr, clientIP, agent},
+			Response: Res{codes.OK},
+			Segment:  Seg{nil, false},
+		},
+		"with-parent": {
+			Trace:    Tra{traceID, spanID, parentID},
+			Request:  Req{remoteAddr, clientIP, agent},
+			Response: Res{codes.OK},
+			Segment:  Seg{nil, false},
+		},
+		"error": {
+			Trace:    Tra{traceID, spanID, ""},
+			Request:  Req{remoteAddr, clientIP, agent},
+			Response: Res{codes.Unknown},
+			Segment:  Seg{status.Error(codes.Unknown, "error"), true},
+		},
+		"fault": {
+			Trace:    Tra{traceID, spanID, ""},
+			Request:  Req{remoteAddr, clientIP, agent},
+			Response: Res{codes.InvalidArgument},
+			Segment:  Seg{status.Error(codes.InvalidArgument, "error"), true},
+		},
+	}
+	for k, c := range cases {
```

Comment:
for new tests like this, I'd recommend using `t.Run` within the for loop:
```go
t.Run(k, func(t *testing.T) {
```
That way you don't need to worry about `k` in every error message.

Reply:
Agree! I also follow the above pattern usually. But got a bit lazy when copy-pasting stuff over.

Reply:
Changed the tests. Thanks!